### PR TITLE
CT-2404 Replace empty name with subject on closed case table

### PR DESCRIPTION
**Pull request by digitalali-moj**
## Self-review checklist

* [x] (1) Quick stakeholder demo done OR
* [x] (2) ...bug with before and after screenshots
* [x] (3) Tests passing
* [x] (4) Branch ready to be merged (not work in progress)
* [x] (5) No superfluous changes in diff
* [x] (6) No TODO's without new ticket numbers
 
### Screenshots
New:
image

 ### Ticket
https://dsdmoj.atlassian.net/browse/CT-2404
 
### Deployment
n/a
 
### Manual testing instructions
Create Offender SAR case with no represeantative name. Then keep moving status until closed. View closed cases table and ensure that subject name is written twice in table. Then try with different representative name, now it should have representative in closed cases table with subject.

### CODE REVIEW

**Review by mseedat-moj — CHANGES_REQUESTED** [review]

File: app/views/cases/filters/closed.html.slim [hunk 1]
```diff
@@ -36,7 +36,7 @@
               = kase.trigger_case_marker
             td aria-label="#{t('.name-subject')}"
               strong
-                = kase.name
+                = kase.name == '' ? kase.subject : kase.name
```
[the remark this answers is not in the document]
Comment (mseedat-moj) [reply app/views/cases/filters/closed.html.slim]: Apologies forgot to actually press the 'submit' button!

File: spec/decorators/case/base_decorator_spec.rb [hunk 2]
```diff
@@ -438,4 +438,39 @@
...
+        @subject
+      end
+    end
+
+    context 'when case represenatative name is empty' do
+      it 'returns the subject as the represenatative' do
+        kase = MockCase.new(
+          representative_name: '',
+        )
+        expect(closed_case_name(kase)).to eq 'Monalisa Khan'
```
Comment (digitalali-moj) [inline spec/decorators/case/base_decorator_spec.rb]: Not quite working yet, WIP

Comment (digitalali-moj) [reply spec/decorators/case/base_decorator_spec.rb]: Working now after patch

**Review by mseedat-moj — APPROVED** [review]
Well done

**Pull request merged**